lib/jose-jws-verify.js

```diff
@@ -183,7 +185,13 @@ JoseJWS.Verifier.prototype.verify = function () {
     if (keyfinder) {
       key_promises[kid] = keyfinder(kid);
     }
-    promises.push(that.cryptographer.verify(sig.aad, that.payload, sig.signature, key_promises[kid], kid));
+    promises.push(that.cryptographer.verify(sig.aad, that.payload, sig.signature, key_promises[kid], kid)
+      .then(function (vr) {
+        if (vr.verified) {
+          vr.payload = Utils.Base64Url.decode(that.payload);
+        }
```

Comment:
I can't think of anything meaningful *to* do. Users are protected from accidentally trusting invalid payloads by not having anything available.